1-Draft/RFCNNNN-Multi-Line-Continuation.md

```diff
@@ -0,0 +1,301 @@
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+`@
+```
+
+The best part is that it doesn't appear this syntax would introduce a breaking
+change at all.
+
+If people feel the backtick is still not visible enough here (and therefore not
+desirable for this purpose), we should keep the `@` portion of the enclosures
+so that we can avoid breaking changes, and simply replace the backtick with
+something else. For example, we could do this instead:
+
+```PowerShell
+Get-ChildItem @-
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+-@
+```
+
+Backticks offer the advantage of continuing what they represent in PowerShell
+already (line continuation), and they are syntactically very similar to here-
+strings when used with the `@` symbol. That last point could be seen as a
+disadvantage as well, because they may be visually harder to distinguish from
+a single-quoted here-string.
+
+The `@-|-@` alternative doesn't pick up on the backtick for line continuation,
+but it is visually unique and easy to see in a script. The `-` character could
+be seen as representing the line that makes up the statement as well.
+
+### Inline splatting
+
+There has also been some discussion about the idea of inline splatting, using a
+format like `-@{...}` or `-@(...)`. Inline splatting has also been discussed
+separately on [RFC0002: Generalized Splatting](https://github.com/PowerShell/PowerShell-RFC/blob/master/2-Draft-Accepted/RFC0002-Generalized-Splatting.md), but using the syntax `@@{...}` or
+`@@(...)`.
+
+Here is an example showing what that might look like:
+
+```PowerShell
+Get-ChildItem -@{
+    LiteralPath = $rootFolder
+    File = $true
+    Filter = '*.ps*1'
+}
+```
+
+Using inline splatting to be able to span a single command across multiple
+lines like this has several limitations, including:
+
+1. You cannot transition to/from the inline splatted syntax without a bunch of
+manual tweaks to the command (either converting parameter syntax into hashtable
+or array syntax or vice versa).
+1. You're forced to choose between named parameters or positional
+parameters/arguments for each splatted collection. i.e. You can splat in a
+hashtable of named parameter/value pairs or an array of positional values, but
+you can't mix the two (the example shown just above is also used earlier in
+this RFC with positional parameters and switch parameters used without values,
+matching the way it is often used as a single-line command).
+1. There's no way to include unparsed arguments after the stop-parsing sigil in
+splatting. You can add it afterwards, but not include it within.
+1. Splatting requires a different syntax than typical parameter/argument input,
+which is more to learn. In contrast, the proposal above only requires learning
+about the `@` sigil (borrowed from splatting, but without specifying hashtables
+or arrays -- just allow all content until a newline), reducing the learning
+curve and allowing users to use parameters the same way in either case.
```

Comment:
This is mostly the same point as your first in this list, and we think usage and discoverability could be fixed in tooling. Furthermore, new operators are notoriously difficult to discover, so I'm not convinced that you wouldn't be in the same boat with a new sigil.

Reply:
Splatting and multi-line continuations are really distinct use cases, and you shouldn't have to use the former to achieve the latter: 

* (a) If I want to specify my _known_ parameters _as usual_, but across multiple lines in a readable fashion, there should be an easy way to do that, and the trailing `@` offers just that.

* (b) If I want to construct the (variable) set of parameter values _programmatically_, I'll use splatting.

(a) strikes me as far more common than (b)

Reply:
Another way of looking at it: if we had good support for (a), as proposed, then perhaps we wouldn't need the _inline_ splatting variant at all, as its primary purpose seems to be to (indirectly) address (a)

Reply:
> Another way of looking at it: if we had good support for (a), as proposed, then perhaps we wouldn't need the _inline_ splatting variant at all, as its primary purpose seems to be to (indirectly) address (a)

Exactly this. Inline splatting feels like the wrong solution to the problem. Splatting's intent is not for code readability.